### PR TITLE
Migrate Enzyme to RTL - src/app/machines/views/MachineDetails/MachineNetwork *Part 1*

### PR DESCRIPTION
**Pull request by ndv99**
## Done

- Migrated the following tests to RTL:
  - [src/app/machines/views/MachineDetails/MachineNetwork/AddAliasOrVlan/AddAliasOrVlan.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_machineNetwork?expand=1#diff-dfd6eb866d568cbc19f85622843fc8060ee5539be80930719855ecba0252b2b3)
  - [src/app/machines/views/MachineDetails/MachineNetwork/AddAliasOrVlan/AddAliasOrVlanFields/AddAliasOrVlanFields.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_machineNetwork?expand=1#diff-359a7de789c4179dad73a78bfb02c513a7d1c0d91df1e8ba45749a43b9f77dff)
  - [src/app/machines/views/MachineDetails/MachineNetwork/AddBondForm/AddBondForm.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_machineNetwork?expand=1#diff-57de1e6c8196b02cdba060335e17e9d0b0cf90143abe631e8fade8d203b87dd4)
  - [src/app/machines/views/MachineDetails/MachineNetwork/AddBridgeForm/AddBridgeForm.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_machineNetwork?expand=1#diff-33a2e88488a91095add4b6e472891639d3cc959bbc07447fedff551ce306ff92)
  - [src/app/machines/views/MachineDetails/MachineNetwork/AddInterface/AddInterface.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_machineNetwork?expand=1#diff-a8a2955a6916685a272a16f153f212583b99570909fe1f6fe63822ca0c6d64cf)
  - [src/app/machines/views/MachineDetails/MachineNetwork/BondForm/BondFormFields/BondFormFields.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_machineNetwork?expand=1#diff-7cc07e59ad48e7a274c298e042af620da6fb35293c720cb346711d20731acfaf)

- Fixed MAC address formatting in testing/factories/nodes.ts

## Related issues

Related to https://github.com/canonical/app-tribe/issues/1347

## Fixes

Fixes https://github.com/canonical/app-tribe/issues/1343
Fixes https://github.com/canonical/app-tribe/issues/1353
Fixes https://github.com/canonical/app-tribe/issues/1344
Fixes https://github.com/canonical/app-tribe/issues/1088

### CODE REVIEW

File: src/app/machines/views/MachineDetails/MachineNetwork/AddAliasOrVlan/AddAliasOrVlan.tsx [hunk 1]
```diff
@@ -30,6 +30,11 @@ import type { NetworkInterface } from "app/store/types/node";
 import vlanSelectors from "app/store/vlan/selectors";
 import { preparePayload } from "app/utils";
 
+export enum Labels {
+  SubmitLabel = "Save interface",
+  SecondarySubmitLabel = "Save and add another",
```
Comment (petermakowski) [inline src/app/machines/views/MachineDetails/MachineNetwork/AddAliasOrVlan/AddAliasOrVlan.tsx:35]: Might be good to use more specific labels here.

File: src/app/machines/views/MachineDetails/MachineNetwork/AddInterface/AddInterface.test.tsx [hunk 2]
```diff
@@ -49,26 +48,27 @@ describe("AddInterface", () => {
...
-            <AddInterface close={jest.fn()} systemId="abc123" />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
+    renderWithBrowserRouter(
+      <AddInterface close={jest.fn()} systemId="abc123" />,
+      { route, wrapperProps: { store } }
     );
-    await waitForComponentToPaint(wrapper);
+    // await waitForComponentToPaint(wrapper);
```
Comment (petermakowski) [inline src/app/machines/views/MachineDetails/MachineNetwork/AddInterface/AddInterface.test.tsx:71]: 👀

File: src/app/machines/views/MachineDetails/MachineNetwork/AddInterface/AddInterface.test.tsx [hunk 3]
```diff
@@ -80,49 +80,52 @@ describe("AddInterface", () => {
...
+    //     fabric: 1,
+    //     ip_address: "1.2.3.4",
+    //     mac_address: "28:21:c6:b9:1b:22",
+    //     mode: NetworkLinkMode.LINK_UP,
+    //     name: "eth1",
+    //     subnet: 1,
+    //     tags: ["a", "tag"],
+    //     vlan: 1,
+    //   })
+    // );
```
Comment (petermakowski) [inline src/app/machines/views/MachineDetails/MachineNetwork/AddInterface/AddInterface.test.tsx:109]: 👀

File: src/app/machines/views/MachineDetails/MachineNetwork/BondForm/BondFormFields/BondFormFields.tsx [hunk 4]
```diff
@@ -94,6 +94,7 @@ const BondFormFields = ({ selected, systemId }: Props): JSX.Element | null => {
           value={MacSource.NIC}
         />
         <FormikField
+          aria-label="Select bond member"
```
Comment (petermakowski) [inline src/app/machines/views/MachineDetails/MachineNetwork/BondForm/BondFormFields/BondFormFields.tsx:97]:
```suggestion
          aria-label="bond member"
```

File: src/testing/factories/nodes.ts [hunk 5]
```diff
@@ -167,7 +167,7 @@ export const networkInterface = extend<Model, NetworkInterface>(model, {
   link_connected: true,
   link_speed: 10000,
   links: () => [],
-  mac_address: (i: number) => `00.00.00.00.00.${i}`,
+  mac_address: (i: number) => `00:00:00:00:00:${i}`,
```
Comment (petermakowski) [inline src/testing/factories/nodes.ts:170]: 👍

File: src/app/machines/views/MachineDetails/MachineNetwork/AddAliasOrVlan/AddAliasOrVlanFields/AddAliasOrVlanFields.test.tsx [hunk 6]
```diff
@@ -1,16 +1,14 @@
...
 
 import AddAliasOrVlanFields from "./AddAliasOrVlanFields";
 
 import type { RootState } from "app/store/root/types";
 import { NetworkInterfaceTypes } from "app/store/types/enum";
 import { rootState as rootStateFactory } from "testing/factories";
+import { renderWithBrowserRouter } from "testing/utils";
 
-const mockStore = configureStore();
+const route = "/machines";
```
Comment (petermakowski) [inline src/app/machines/views/MachineDetails/MachineNetwork/AddAliasOrVlan/AddAliasOrVlanFields/AddAliasOrVlanFields.test.tsx:11]: Just noticed one more thing. 
We have a constant for this which you can import:
`import urls from "app/base/urls";`
```suggestion
const route = urls.machines.index;
```

Comment (ndv99) [reply src/app/machines/views/MachineDetails/MachineNetwork/AddAliasOrVlan/AddAliasOrVlanFields/AddAliasOrVlanFields.test.tsx]: Can't believe I forgot about that! Changed it now

**Review by petermakowski — APPROVED** [review]

**Pull request merged**